test/dsets.c

```diff
@@ -5683,6 +5683,306 @@ test_multiopen(hid_t file)
+
+        TESTING("    double (setup)");
+
+        /* Define user-defined double-precision floating-point type for dataset */
+        datatype = H5Tcopy(H5T_IEEE_F64BE);
+        if (H5Tset_fields(datatype, (size_t)55, (size_t)46, (size_t)9, (size_t)5, (size_t)41) < 0)
+            goto error;
+        offset = 5;
+        if (H5Tset_offset(datatype, offset) < 0)
+            goto error;
+        precision = 51;
+        if (H5Tset_precision(datatype, precision) < 0)
+            goto error;
+        if (H5Tset_size(datatype, (size_t)8) < 0)
+            goto error;
+        if (H5Tset_ebias(datatype, (size_t)255) < 0)
+            goto error;
+
+        /* Create the data space */
+        if ((space = H5Screate_simple(2, size, NULL)) < 0)
+            goto error;
+
+        /* Create the dataset */
+        if ((dataset =
+                 H5Dcreate2(file, "double_type", datatype, space, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT)) < 0)
+            goto error;
+
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * STEP 1: Test writing to it.
+         *----------------------------------------------------------------------
+         */
+        TESTING("    double (write)");
+
+        if (H5Dwrite(dataset, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, orig_data) < 0)
+            goto error;
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * STEP 2: Try to read the data we just wrote.
+         *----------------------------------------------------------------------
+         */
+        TESTING("    double (read)");
+
+        /* Read the dataset back */
+        if (H5Dread(dataset, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, new_data) < 0)
+            goto error;
+
+        /* Check that the values read are the same as the values written
+         * Assume size of long long = size of double
+         */
+        for (i = 0; i < (size_t)size[0]; i++) {
+            for (j = 0; j < (size_t)size[1]; j++) {
+                if (HDisnan(orig_data[i][j]))
+                    continue; /* skip if value is NaN */
+                if (!H5_DBL_ABS_EQUAL(new_data[i][j], orig_data[i][j])) {
+                    H5_FAILED();
+                    printf("    Read different values than written.\n");
+                    printf("    At index %lu,%lu\n", (unsigned long)i, (unsigned long)j);
+                    goto error;
+                }
+            }
+        }
+
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * Cleanup
+         *----------------------------------------------------------------------
+         */
+        if (H5Tclose(datatype) < 0)
+            goto error;
+        if (H5Sclose(space) < 0)
+            goto error;
+        if (H5Dclose(dataset) < 0)
+            goto error;
+    }
+#if 0
+#if H5_SIZEOF_LONG_DOUBLE != H5_SIZEOF_DOUBLE
```

Comment:
#if 0 because this is the test that needs fixed - the orig_data values need to be appropriate for long double (they are just copies of double from above).